test/helper.lua

```diff
@@ -258,25 +258,27 @@ function helpers.tarantool_version()
     return major, minor, patch
 end
 
-function helpers.vinyl_is_broken()
-    -- Blocked by https://github.com/tarantool/tarantool/issues/6448
+function helpers.vinyl_is_supported()
     local major, minor, patch = helpers.tarantool_version()
 
-    -- Since Tarantool 1.10.11.
-    local broken_v1_10 = major >= 1 and (minor > 10 or minor == 10 and patch >= 11)
-
-    -- Tarantool >= 2.1.0 and < 2.8.2.
-    local broken_v2 = (major >= 2 and major < 3) and (minor >= 1 and minor < 8 or minor == 8 and patch <= 2)
-
-    return broken_v1_10 or broken_v2
+    -- Blocked by https://github.com/tarantool/tarantool/issues/6448
+    -- Tarantool < 1.10.11 for Tarantool v1
+    -- Tarantool <= 2.1.0 or >= 2.10 for Tarantool v2
+    return (major == 1 and minor < 10) or
+        (major == 1 and minor == 1 and patch < 11) or
+        (major == 2 and minor <= 1) or
+        (major == 2 and minor >= 10) or
+        (major >= 3)
 end
 
-function helpers.memtx_func_index_is_broken()
+function helpers.memtx_func_index_is_supported()
     local major, minor, patch = helpers.tarantool_version()
-    if major > 2 or (major == 2 and minor == 8 and patch >= 4) or (major == 2 and minor >= 10) then
-        return false
-    end
-    return true
+
+    -- Blocked by https://github.com/tarantool/tarantool/issues/6786
+    -- Tarantool >= 2.8.4 or Tarantool >= 2.10
+    return (major == 2 and minor == 8 and patch >= 4) or
+        (major == 2 and minor >= 10) or
+        (major >= 3)
```

Comment:
I think a short comment is needed here. The following facts may be not obvious for a reader:

* Functional indexes for memtx storage engine are introduced in 2.2.1. The 1.10 series does not support them at all.
* Vinyl is not supported yet: https://github.com/tarantool/tarantool/issues/4492.
* The https://github.com/tarantool/tarantool/issues/6786 problem was fixed in 2.8.4 and 2.10.0. We verify the related functionality only on tarantool versions that include this fix.

Reply:
Thank you, I've added more comments.